libcoveweb2/utils.py

```diff
@@ -2,7 +2,34 @@ def group_data_list_by(data_list, get_key_function):
     out = {}
     for item in data_list:
         key = get_key_function(item)
-        if not key in out:
+        if key not in out:
             out[key] = []
         out[key].append(item)
     return out
+
+
+def get_file_type(file_name):
+    """Takes an filename (type string), and returns a string saying what type it is."""
+    # Older versions of this could take DJango objects to.
+    # Tho we don't really want to support that, put in a check for that.
+    if not isinstance(file_name, str) and hasattr(file_name, "path"):
+        file_name = file_name.path
+    # First, just check the extension on the file name
+    if file_name.lower().endswith(".json"):
```

Comment:
Does this need to capture `.geojson` as well now, assuming that they won't always end with `.json`?

Reply:
Wait, does the try catch block handle that?

Reply:
Not urgently, cos that's actually only used for getting type of spreadsheet - but I'll note for later if that's ok?